### ISSUE TITLE
Timing issue - I can save end time before the start time

### ISSUE THREAD

**Issue by McDKirra** [mikezimm/TrackMyTime7#41]
Timing issue - I can save end time before the start time : by this I mean that if I start at 5pm, it allows me to select the end time to be prior to the start time. :) Could you please have a check?

**Comment by mikezimm** [comment]
I am pushing a new update which should resolve this one.

The problem was that when the time was "empty" ie when first loading the webpart,
The react control for the time picker would be nice enough to put "now" in the picker.
However, this did not update the formEntry in state which is what actually gets saved when pressing save.

To solve, I added a couple things:

updated _updateEntryType as follows:
![image](https://user-images.githubusercontent.com/49648086/74465904-3e787800-4e64-11ea-8bc1-54f8808019e6.png)

I also squashed another bug in the addThisItemToState...
This update will update the times on the form to "now" if the save is successful so that if you hit save again, it would not have old times which could lead to double entries.

![image](https://user-images.githubusercontent.com/49648086/74465982-5d770a00-4e64-11ea-970d-ad14b477b622.png)
